### ISSUE TITLE
Lye liquid>Lye powder recipe hopelessly inefficient

### ISSUE THREAD

**Issue by Solusphere** [CleverRaven/Cataclysm-DDA#22406]
If you want to make Lye powder, there are two options. One is a recipe found in chemistry books, which uses charcoal and is fairly sensible. This issue concerns the other one, which is autolearned through cooking and uses liquid Lye.

To make 500ml of Lye, you need 50 wood ash. This can be crafted in batches of 10, taking 15 minutes each. When you have the ash, crafting the Lye takes an hour, making a total crafting time of 2 hours and 15 minutes for 500ml.

Crafting a single batch of Lye powder requires 40 units of lye, which would need 86 hours of nonstop crafting to achieve, and at least 200 two-by-fours or 1000 splintered wood.

I know the autolearned recipes are supposed to be less efficient than the book ones, but being nearly 50 times less efficient seems a little overboard.

**Comment by Coolthulhu** [comment]
The ash making process is currently barely a step above a placeholder.
What it should be doing is creating ash on any burned wood object, not just a specific "ash crafting recipe" or manual shoveling.

**Comment by Solusphere** [comment]
That would help significantly, but even if making ash was instant this would still represent a 40+ hour crafting time per batch.

**Comment by atealein** [comment]
I agree that the recipe for lye powder from lye (from ash) should be balanced better. For one it should allow the production of more lye powder from lye; Another thing is that the batch quantity should require less time if done in large batches; and possibly scale up the quantities in the base recipe.

**Comment by ituluwituluwzev** [comment]
I added the lye->powder recipe. It is indeed too time consuming to be practical, whether you craft the ash or burn forest and collect ash. The ratio is correct, as one can see in the soap recipe:[ "lye_powder", 10 ], [ "lye", 2 ].

The solution is to lower the stack-size of lye powder, which I myself don't know how to.

**Comment by Coolthulhu** [comment]
I'm pretty sure there is a way to have a recipe produce smaller/bigger stacks than default charges.

**Comment by ituluwituluwzev** [comment]
That'd be very useful!

**Comment by Treah** [comment]
Yeah me myself have done this and making lye powder is a drag. I opted to burn down a forest to get the ash needed but still took 40 hours of crafting to make a single batch of powder. 

This could be better adjusted by either making the recipe for lye make more lye or by reducing the crafting time. I like reducing the time better as you can better control how much you want if its in increments of 1. Suggestion would be to reduce it from an hour to 15min. 
Or if you want to be super realistic you could make a new item Ash Water that has to brew for a day or so to turn into lye :)

**Comment by Night-Pryanik** [comment]
> Suggestion would be to reduce it from an hour to 15min.

I've read that making lye that way requires 3 or more hours, so now-present 1 hour is a highly lowered time.

**Comment by AlecWhite** [comment]
Then balance or realism?

**Comment by atealein** [comment]
All of the recipes are lowered (unless you think that making most of the clothing requires 30 minutes when you don't have a sowing machine, in which case I have to burst your bubble) in time from reality to the game. I don't mind you making the lye recipe fixed at 3 hours, but you should then make all the extra batches done at the same time take "no" extra time, since in reality you boil all the things at the same time, not consecutively.

**Comment by atealein** [comment]
Now, going back to the soap recipe (where the lye powder to lye ratio was taken from) and the lye from ash and water recipes... quick search on the internet shows that those are kinda off. It takes (example link, there are literally hundreds of recipes like this one explaining it in detail: https://farmingmybackyard.com/homemade-soap-from-ashes/ ). It takes 10 cups of ashes and about 2 gallons of water to make a gallon of lye water. And following the recipe later on, this guy was using two gallons of lye water for his batch of soaps (however large it was is unclear). 

Other calculators show you exactly how much lye (in powder form and then 1:2 with water as liquid lye) one needs for certain amount of soap. I took our standard soap bar (10) which weights 0.9 kg and played a bit in this tool - https://www.brambleberry.com/Pages/Lye-Calculator.aspx to see how much lye i will need for a solid soap made only with lard. 600 gr lard + 83 gr lye + 200 gr of water obviously made close to 900 gr of soap. 

Is our current soap recipe realistic? 1 unit of lye powder = 1 gram. 1 unit of lard = 130 gr. 1 unit of water = 250 gr. So current recipe (fat + lye powder + water) is kinda on point for the lye powder and water, but underestimates the amount of fat needed by a factor of ~2.5. so might need to be 5 lard instead of 2. However, looking at the same recipe where it uses lye (liquid lye), it goes totally wrong. Using the 1:2 ratio of converting lye powder + water to liquid lye, we are grossly overestimating the amound of liquid lye needed in this recipe. 1 unit of liquid lye is 250 gr, which would be approximately 80 gr lye powder and 170 gr of water. Or kinda 8 time more lye powder than if using lye powder directly.

I hope you see my argument now - the lye powder from liquid lye recipe is based on the ratios in the soap recipe, which are totally wrong. If you then scale down lye powder to be made not from 40 liquid lye units, but 5 (8 times smaller amount) suddenly things become much more reasonable. 5 units of lye will take 250 wood ash to make and (if the batch recipe time scaling remains unchanged) 5 hours of crafting time + 1.5 hours for the lye powder  to produce 200 units of lye powder.. which a more skilled character can create with 50 charcoal and 3 hours.

**Comment by atealein** [comment]
Now, going back to the liquid lye from wood ash recipe: is it realistic? 1 unit of wood ash = 10 gr, 1 unit of water = 250 gr, 1 unit of liquid lye = 250 gr. The site about soap making I referenced earlier gave a clear estimate - 10 cups of wood ash + 2 gallons of water = 1 gallon of liquid lye (at soap-making strength). I will go assume wood ash has similar consistency and density to flour so 10 cups would be approximately equal to 1200 gr of wood ash or 120 units. 2 gallons water would be 30 units, 1 gallon of liquid lye will be 15 units. So real life recipe is 120 wood ash + 30 water = 15 lye. Game recipe is 50 wood ash + 1 water = 1 lye. Add that in real life burning a whole tree will definitely leave more than 100 grams of wood ash and you will notice how this recipe is also very unbalanced towards a material (wood ash) that is not so straightforward to obtain in large quantities. If we have to correct the game recipe to real life it would look something like 8 wood ash + 2 water = 1 lye. 

So basically we have wrong situation here where we have unbalanced recipes A and C and we are discussing the balance of recipe B that is derived from them and links them. All three have to be addressed at the same time.

**Comment by Night-Pryanik** [comment]
@atealein You definitely should open a PR with your suggestions about ash, lye and soap making.

**Comment by atealein** [comment]
Unfortunately I am too new to git and don't have the code compiled locally either - I download the precompiled installation packages of the game :S So I can just give suggestions and hope a good and hardworking soul will take the torch and code them in ;)

**Comment by MattD94** [comment]
Another thing that might be worth considering to help out with making lye would be a furniture item that you add ash and water to and will automatically make lye over several hours.

This site goes into detail of making a leaching barrel for lye if anyone is interested [https://www.motherearthnews.com/homesteading-and-livestock/how-to-make-soap-from-ashes-zmaz72jfzfre](url)

Also lye made from wood ash is actually a potassium hydroxide, not sodium hydroxide, like the in game description says.

**Comment by Treah** [comment]
I noticed in a current game that burning down trees in a forest no longer yields ash.  I wouldn't mind making the changes described by atealein even if they are a bit old since they are simply json changes.

**Comment by DaviBones** [comment]
This should probably be closed now, was fixed in #23743.
